app/views/miq_request/_request_dialog_details.html.haml

```diff
@@ -28,12 +28,15 @@
       = h(field.values.detect { |k, _v| k == wf.value(field.name) }.try(:last) || wf.value(field.name))
 
     - when "DialogFieldDropDownList"
-      = h(field.value.blank? ? _("<None>") : field.value.to_s.gsub("\x1F", ", "))
+      - if field.value.instance_of?(Array)
+        = h(field.value.blank? ? _("<None>") : h(field.value.join(", ")))
+      - else
+        = h(field.value.blank? ? _("<None>") : h(field.value))
 
     - when 'DialogFieldTagControl'
-      - value = wf.value(field.name) || '' # it returns in format Clasification::id
-      - classifications = value.split(',').map do |c|
-        - classification = Classification.find_by(id: c.split('::').second)
+      - value = wf.value(field.name) || ''
+      - classifications = value.map do |c|
+        - classification = Classification.find_by(:id => c)
```

Comment:
Find in a loop is an N+1 expensive operation. Instead this should be moved out of the loop and worried in a single query.